src/otaclient/main.py

```diff
@@ -265,9 +281,25 @@ def main() -> None:  # pragma: no cover
         )
         _otaproxy_control_t.start()
 
+    _stop_request_thread = threading.Thread(
+        target=partial(
+            stop_request_thread,
+            otaclient_main_queue=otaclient_main_queue,
+            critical_zone_flags=critical_zone_flag,
+        ),
+        daemon=True,
+        name="otaclient_stop_request_thread",
+    )
+
+    _stop_request_thread.start()
+
     while True:
         time.sleep(HEALTH_CHECK_INTERVAL)
 
+        if not _stop_request_thread.is_alive():
```

Comment:
From my point of view, I refrain using this condition to judge whether stop is requested. `Stop thread is dead` equals to `stop requested` sounds very unnatural and counter-intuitive  to me.
Let's use a `stop_request_accepted` threading.Event instead.